src/compas_ghpython/artists/base.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+import compas_rhino
+
+
+__all__ = ["BaseArtist"]
+
+
+_ITEM_ARTIST = {}
+
+
+class BaseArtist(object):
+    """Base class for all artists.
+
+    Attributes
+    ----------
+    guids : list
+        A list of the GUID of the Rhino objects created by the Grasshopper artist.
+
+    """
+
+    def __init__(self):
+        self.guids = []
+
+    @staticmethod
+    def register(item_type, artist_type):
+        _ITEM_ARTIST[item_type] = artist_type
+
+    @staticmethod
+    def build(item, **kwargs):
+        """Build an artist corresponding to the item type.
+        Parameters
+        ----------
+        kwargs : dict, optional
+            The keyword arguments (kwargs) collected in a dict.
+            For relevant options, see the parameter lists of the matching artist type.
+        Returns
+        -------
+        :class:`compas_ghpython.artists.BaseArtist`
+            An artist of the type matching the provided item according to an item-artist map.
+            The map is created by registering item-artist type pairs using ``~BaseArtist.register``.
+        """
+        artist_type = _ITEM_ARTIST[type(item)]
+        artist = artist_type(item, **kwargs)
+        return artist
+
+    def draw(self):
+        raise NotImplementedError
+
+    def redraw(self):
+        compas_rhino.rs.EnableRedraw(True)
+
+    def clear(self):
```

Comment:
I am not sure if this is still a draft PR (considering the `NotImplementedError` right above), but GH artists are different from Rhino ones. These artists never add objects to the document, instead the `draw` methods return geometry instances, so, by definition, they have no `clear` method.

Reply:
thanks for explaining. I will fix it and follow the contribution guidelines